### PR TITLE
Verify we can load the role with the correct ID.

### PR DESCRIPTION
**Pull request by RoySegall**
I got this while working on another issue. It seems the role is being saved with `--foo` and not `foo` because i'm checking if there any group the role belong to.

### CODE REVIEW

**Comment by RoySegall** [comment]
@amitaibu Ready for review.

**Comment by pfrenssen** [comment]
An empty file `src/OgRolesHandler.php` is added. We can also maybe rebase this to remove the "Foo" commit and its reversal.

From reading the code I do not really understand the problem. The description and title mention to verify the ID when loading a role, but the majority of the changes in the code and tests seem to be related to making sure the type and bundle are set during saving. I'll look a bit closer now.

File: tests/src/Kernel/Entity/OgRoleTest.php [hunk 1]
```diff
@@ -41,9 +42,23 @@ public function testRoleCreate() {
...
+      $this->assertTrue(TRUE, 'OG role without bundle/group was not saved.');
+    }
+
+    $og_role
+      ->setGroupType('node')
+      ->setGroupBundle('group')
       ->save();
 
+    $this->assertNotEmpty(OgRole::load('node-group-content_editor'), 'The role was created with the expected ID.');
+
```
Comment (pfrenssen) [inline tests/src/Kernel/Entity/OgRoleTest.php:61]: Great, we were missing this!

File: tests/src/Kernel/Entity/OgRoleTest.php [hunk 2]
```diff
@@ -69,9 +86,10 @@ public function testRoleCreate() {
       ->setLabel('Content editor')
       ->setGroupType('entity_test')
       ->setGroupBundle('group')
+      ->setGroupID(1)
       ->save();
 
-    $this->assertEquals('entity_test-group-content_editor', $og_role->id());
+    $this->assertEquals('entity_test-group-1-content_editor', $og_role->id());
```
Comment (pfrenssen) [inline tests/src/Kernel/Entity/OgRoleTest.php:93]: OK I get it now. This tests that if a group ID is set, then the generated role ID contains the group ID. The negative case (no group ID set) is tested in the previous test.

Am I right in assuming that a group ID is optional, and is only needed when you are creating two roles of the same type/bundle pair?

**Comment by pfrenssen** [comment]
This is looking good to me. I would still rebase the branch to remove the "Foo" commit and its revert + make sure the empty file is not generated.

**Comment by amitaibu** [comment]
Merged (squash). Thanks.

**Pull request closed without merging**